### PR TITLE
Music sync mod

### PR DESCRIPTION
**Pull request by myst6re**

### CODE REVIEW

**Review by julianxhokaxhiu — APPROVED** [review]
LGTM. Can I proceed with the merge or is there anything else to be made on top?

**Comment by myst6re** [comment]
> LGTM. Can I proceed with the merge or is there anything else to be made on top?

I need to fix a problem with worldmap and I hope optimize the seeking

File: src/music.cpp [hunk 1]
```diff
@@ -211,40 +212,57 @@ bool play_music(const char* music_name, uint32_t music_id, int channel, NxAudioE
...
-						playing = nxAudioEngine.playMusic(music_name, music_id, channel, options);
+					playing = nxAudioEngine.playMusic(battle_name, music_id, channel, options);
 				}
-				else
-					// Not in field, switch back to default
-					playing = nxAudioEngine.playMusic(music_name, music_id, channel, options);
+			}
+			else
+			{
+				ffnx_warning("%s: Unknown battle_id\n", __func__);
```
Comment (julianxhokaxhiu) [inline src/music.cpp:259]: Shall we maybe put this behind a `trace_*` flag? What do you think?

File: src/audio.cpp [hunk 2]
```diff
@@ -629,9 +645,13 @@ bool NxAudioEngine::playMusic(const char* name, uint32_t id, int channel, MusicO
 		// Keep audioSource pointer somewhere to delete it after musicHandle is stopped
 		_audioSourcesToDeleteLater.push_back(NxAudioEngineMusicAudioSource(music.handle, audioSource));
 
-		if (options.offsetSeconds > 0) {
+		if (options.offsetSeconds > 0 || (options.sync && options.lastMusicOffset > 0)) {
 			if (trace_all || trace_music) ffnx_info("%s: seek to time %fs\n", __func__, options.offsetSeconds);
-			_engine.seek(music.handle, options.offsetSeconds);
+			SoLoud::time offsetSeconds = options.sync ? options.lastMusicOffset : options.offsetSeconds;
+			if (musicLength > 0) {
+				offsetSeconds = std::fmod(offsetSeconds, musicLength);
```
Comment (myst6re) [inline src/audio.cpp:652]: Oh no I though I removed this

**Comment by julianxhokaxhiu** [comment]
Mark it as final when you think is ready to be merged. Thank you.

**Comment by myst6re** [comment]
Ok ready

**Comment by myst6re** [comment]
Fewtch22 have new feature requests:
 - add ability to customise the music with the field music id (those visible in the flevel to start a music via an opcode)
 - adding a way to disable a music (and not stop the previous one)

**Review by julianxhokaxhiu — APPROVED** [review]

**Comment by julianxhokaxhiu** [comment]
Thank you!

**Pull request merged**